src/main/java/io/fabric8/maven/docker/config/handler/compose/DockerComposeServiceWrapper.java

```diff
@@ -182,11 +182,22 @@ NetworkConfig getNetworkConfig() {
             NetworkConfig ret = new NetworkConfig(NetworkConfig.Mode.custom, custom);
             Object aliases = toJoin.get(custom);
             if (aliases != null) {
-                if (!(aliases instanceof List)) {
-                    throwIllegalArgumentException("'networks:' Aliases must be given as a list of string");
-                }
-                for (String alias : (List<String>) aliases) {
-                    ret.addAlias(alias);
+                if (aliases instanceof List) {
+                    for (String alias : (List<String>) aliases) {
+                        ret.addAlias(alias);
+                    }
+                } else if (aliases instanceof LinkedHashMap) {
+                    LinkedHashMap<String, ArrayList<String>> map = (LinkedHashMap<String, ArrayList<String>>) aliases;
+                    if (map.containsKey("aliases")) {
+                        for (String alias : map.get("aliases")) {
+                            ret.addAlias(alias);
+                        }
+                    } else {
+                        throwIllegalArgumentException(
+                                "'networks:' Aliases must be given as a linked has map of strings. 'aliases' key not founded");
```

Comment:
why does it have to be a LinkedHashMap ? Just a "... as a map of strings", so I would use this error message:

```
'networks:' No aliases entry found in network config map
```

Reply:
Message updated